prow/kube/client.go

```diff
@@ -366,23 +372,29 @@ func (c *Client) GetProwJob(name string) (ProwJob, error) {
 	c.log("GetProwJob", name)
 	var pj ProwJob
 	err := c.request(&request{
-		method: http.MethodGet,
-		path:   fmt.Sprintf("/apis/prow.k8s.io/v1/namespaces/%s/prowjobs/%s", c.namespace, name),
+		path: fmt.Sprintf("/apis/prow.k8s.io/v1/namespaces/%s/prowjobs/%s", c.namespace, name),
 	}, &pj)
 	return pj, err
 }
 
 func (c *Client) ListProwJobs(selector string) ([]ProwJob, error) {
 	c.log("ListProwJobs", selector)
-	var jl struct {
-		Items []ProwJob `json:"items"`
+	if c.deckURL == "" {
+		var jl struct {
+			Items []ProwJob `json:"items"`
+		}
+		err := c.request(&request{
+			path:  fmt.Sprintf("/apis/prow.k8s.io/v1/namespaces/%s/prowjobs", c.namespace),
+			query: map[string]string{"labelSelector": selector},
+		}, &jl)
+		return jl.Items, err
 	}
+	var jl []ProwJob
 	err := c.request(&request{
-		method: http.MethodGet,
-		path:   fmt.Sprintf("/apis/prow.k8s.io/v1/namespaces/%s/prowjobs", c.namespace),
-		query:  map[string]string{"labelSelector": selector},
+		deckPath: "/prowjobs.js",
```

Comment:
Can't you append this inside `doRequest` if `deckURL` is specified?

Reply:
FIxed. I had it that old way because k8s says `{"items":[]}` but deck was just saying `[]`. I fixed deck to match k8s.